examples/jsondiff/check_test.go

```diff
@@ -0,0 +1,137 @@
+package jsondiff_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/searis/subtest"
+	pkg "github.com/searis/subtest/examples/jsondiff"
+)
+
+func TestEqualJSON(t *testing.T) {
+	// This test validates that the EqualJSON check functions work; it is not a
+	// direct example of usage.
+
+	const (
+		a = `{"foo": "A", "bar": "B"}`
+		b = `{"foo": "A", "bar": "C"}`
+		c = `{"foo": "A"}`
+	)
+
+	t.Run("when a is cheked against a", func(t *testing.T) {
+		err := pkg.EqualJSON(a)(a)
+		t.Run("then the check should pass", subtest.Value(err).NoError())
+	})
+
+	t.Run("when []byte(a) is checked against a", func(t *testing.T) {
+		err := pkg.EqualJSON(a)([]byte(a))
+		t.Run("then the check should pass", subtest.Value(err).NoError())
+	})
+
+	t.Run("when json.RawMessage(a) is checked against a", func(t *testing.T) {
+		err := pkg.EqualJSON(a)(json.RawMessage(a))
+		t.Run("then the check should pass", subtest.Value(err).NoError())
+	})
+
+	t.Run("when b is cheked against a", func(t *testing.T) {
+		err := pkg.EqualJSON(a)(b)
+		t.Run("then the check should fail", subtest.Value(err).Error())
+	})
+
+	t.Run("when c is cheked against a", func(t *testing.T) {
+		err := pkg.EqualJSON(a)(c)
+		t.Run("then the check should fail", subtest.Value(err).Error())
+	})
+	t.Run("when a is cheked against c", func(t *testing.T) {
+		err := pkg.EqualJSON(c)(a)
+		t.Run("then the check should fail", subtest.Value(err).Error())
+	})
+}
+
+func TestSubsetOfJSON(t *testing.T) {
+	// This test validates that the SubsetOfJSON check functions work; it is not
+	// a direct example of usage.
+
+	const (
+		a = `{"foo": "A", "bar": "B"}`
+		b = `{"foo": "A", "bar": "C"}`
+		c = `{"foo": "A"}`
+	)
+
+	t.Run("when a is cheked against a", func(t *testing.T) {
+		err := pkg.SubsetOfJSON(a)(a)
+		t.Run("then the check should pass", subtest.Value(err).NoError())
+	})
+
+	t.Run("when []byte(a) is checked against a", func(t *testing.T) {
+		err := pkg.SubsetOfJSON(a)([]byte(a))
+		t.Run("then the check should pass", subtest.Value(err).NoError())
+	})
+
+	t.Run("when json.RawMessage(a) is checked against a", func(t *testing.T) {
+		err := pkg.SubsetOfJSON(a)(json.RawMessage(a))
+		t.Run("then the check should pass", subtest.Value(err).NoError())
+	})
+
+	t.Run("when b is checked against a", func(t *testing.T) {
+		err := pkg.SubsetOfJSON(a)(b)
+		t.Run("then the check should fail", subtest.Value(err).Error())
+	})
+
+	t.Run("when c is checked against a", func(t *testing.T) {
```

Comment:
C ⊆ A  and C ⊊ A  is both true, C is a proper subset of A

Reply:
There was a mistake in the test name. `"then the check should fail"` should have been `"then the check should pass"`